### PR TITLE
remove v0.11.3 support; add block_sim to CI

### PR DESCRIPTION
**Pull request by tersec**
Ready to merge at any time.

95% of this collapses/removes the spec-version-dependent conditional compilation blocks.

The only semantic change is that it adds `block_sim` to the CI, for some attestation and block pool integration testing.

### CODE REVIEW

**Comment by arnetheduck** [comment]
let's create a `witti` at the commit before the merge, for historical reference

**Comment by tersec** [comment]
Seems reasonable. Probably immediately after merging to avoid risking incorrect guesses as to which commit that is.

**Comment by tersec** [comment]
Helps with https://github.com/status-im/nim-beacon-chain/issues/364

**Comment by tersec** [comment]
Jenkins failure is unrelated to content of PR, an issue with the machine running the test.

**Comment by arnetheduck** [comment]
> immediately after merging

...tagging the commit before, right? we want one that still works :)

**Comment by tersec** [comment]
> > immediately after merging
> 
> ...tagging the commit before, right? we want one that still works :)

Yes, if if the commits are `A -> B -> [this commit]`, then tag `B` with something like `git tag -a witti ${COMMITHASH_OF_B}`

File: beacon_chain/inspector.nim [hunk 1]
```diff
@@ -197,22 +195,12 @@ func getTopics(forkDigest: ForkDigest,
...
     var topics = newSeq[string](ATTESTATION_SUBNET_COUNT * 2)
     var offset = 0
     for i in 0'u64 ..< ATTESTATION_SUBNET_COUNT.uint64:
-      when ETH2_SPEC == "v0.12.1":
-        topics[offset] = getAttestationTopic(forkDigest, i)
-        topics[offset + 1] = getAttestationTopic(forkDigest, i) & "_snappy"
-      else:
-        topics[offset] = getMainnetAttestationTopic(forkDigest, i)
-        topics[offset + 1] = getMainnetAttestationTopic(forkDigest, i) & "_snappy"
+      topics[offset] = getAttestationTopic(forkDigest, i)
```
Comment (arnetheduck) [inline beacon_chain/inspector.nim:202]: we can remove the non-snappy topics now

Comment (tersec) [reply beacon_chain/inspector.nim:202]: Will do in another PR

**Comment by tersec** [comment]
https://github.com/status-im/nim-beacon-chain/tree/witti

File: beacon_chain/inspector.nim (hunk 1 above)
Comment (tersec) [reply beacon_chain/inspector.nim:202]: https://github.com/status-im/nim-beacon-chain/pull/1261

**Pull request merged**